File: Algorytmy i struktury danych/Python/AiSD5_C2 Ćwiczenie 2/zad2_rok_przestepny.py
def Czytaj_rok():
    rok = int(input("Podaj rok: "))
    return rok;
def Czy_przestepny(rok):
    return ((rok%4==0) and not (rok%100==0)) or (rok % 400 == 0)
def Wypisz_czy_przestepny(rok,przestepny):
    if(przestepny):
        print("Rok", rok, "jest przestepny")
    else:
        print("Rok", rok, "nie jest przestepny")
def Glowna_procedura():
    rok = Czytaj_rok()
    Wypisz_czy_przestepny(rok,Czy_przestepny(rok))

Glowna_procedura()

File: Algorytmy i struktury danych/Pascal/AiSD5_C2 Ćwiczenie 2/zad2_rok_przestepny.pas
program rok_przestepny;

function Czytaj_rok: Integer;
var
   rok: Integer; 
begin
   write('Podaj rok: ');
   readln(rok);
   Czytaj_rok:=rok;
end;

function Czy_przestepny(rok :Integer ): Boolean;
begin
   Czy_przestepny:= ((rok mod 4 = 0) AND (NOT (rok mod 100 = 0)) ) OR (rok mod 400 = 0); 
end;

procedure Wypisz_czy_przestepny(rok : Integer; przestepny: Boolean );
begin
  if przestepny then
     writeln('Rok ', rok, ' jest przestepny')
  else
     writeln('Rok ', rok, ' nie jest przestepny');
end;

procedure Glowna_procedura;
var
   rok        : Integer;
   przestepny : Boolean;
begin
   rok:= Czytaj_rok();
   przestepny:= Czy_przestepny(rok);
   Wypisz_czy_przestepny(rok,przestepny);
end;

begin
   Glowna_procedura()
end.
